azkaban-web-server/src/main/java/cloudflow/daos/ProjectDaoImpl.java

```diff
@@ -0,0 +1,164 @@
+            "Project with name " + name + " already exists.");
+      }
+    } catch (final SQLException ex) {
+      log.error("Check if project '" + name + "' exists failed with: " + ex);
+      throw new ProjectManagerException("Check if project '" + name + "' exists failed");
+    }
+
+    final SQLTransaction<Long> insertAndGetProjectId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_PROJECT, project.getName(), project.getDescription(),
+          project.getSpaceId(), user.getUserId(), currentTime, user.getUserId(), currentTime,
+          1, true, 2, null);
+      transOperator.getConnection().commit();
+      return transOperator.getLastInsertId();
+    };
+
+    String projectId = "";
+    try {
+      projectId = dbOperator.transaction(insertAndGetProjectId).toString();
+      if (projectId.isEmpty()) {
+        throw new ProjectManagerException("Insert project for '" + name + "' failed.");
+      }
+      projectAdminDaoImpl.addAdmins(projectId, project.getAdmins());
+    } catch (SQLException ex) {
+      log.error(INSERT_PROJECT + " failed with exception:", ex);
+      throw new ProjectManagerException("Insert project for '" + name + "' failed.",
+          ex);
+    }
+    return projectId;
+  }
+
+  @Override
+  public Optional<Project> get(String projectId) {
+    List<Project> projects = new ArrayList<>();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+    try {
+      projects = dbOperator.query(FetchProjectHandler.SELECT_PROJECT_WITH_ID, fetchProjectHandler,
+          projectId);
+      for(Project project : projects) {
+        project.setAdmins(projectAdminDaoImpl.findAdminsByProjectId(project.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Select for project record with id " + projectId + " failed: ", ex);
+    }
+    return projects.isEmpty() ? Optional.empty() : Optional.of(projects.get(0));
+  }
+
+  @Override
+  public List<Project> getAll(User user) {
+    List<Project> projects = new ArrayList<>();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+    try {
+      projects = dbOperator.query(FetchProjectHandler.SELECT_ALL_PROJECTS_BY_USER,
+          fetchProjectHandler, user.getUserId(), user.getUserId());
+      for(Project project : projects) {
+        project.setAdmins(projectAdminDaoImpl.findAdminsByProjectId(project.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Get all projects for user '" + user.getUserId() + "'failed: ", ex);
+    }
+    return projects;
+  }
+
+
+  public static class FetchProjectHandler implements ResultSetHandler<List<Project>> {
+
+    static String SELECT_PROJECT_WITH_ID =
+        "SELECT id, name, description, space_id, created_by, creation_time, last_modified_by, "
+            + "last_modified_time, version,  active, enc_type, settings_blob FROM project WHERE id"
```

Comment:
Do we need to add active=1 to the where clause?

Reply:
Added.